packages/terra-pills/src/PillList.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  KEY_SPACE,
+  KEY_RETURN,
+} from 'keycode-js';
+import PropTypes from 'prop-types';
+import classNamesBind from 'classnames/bind';
+import ThemeContext from 'terra-theme-context';
+import styles from './Pill.module.scss';
+import Pill from './Pill';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * The aria label string value for the pills container.
+   */
+  ariaLabel: PropTypes.string.isRequired,
+  /**
+   * The content to be shown in the pills container. Should only contain the sub-component pills
+   */
+  children: PropTypes.node,
+  /**
+   * Determines if the Pill List rolls up or not.
+   */
+  isCollapsible: PropTypes.bool,
+};
+
+const defaultProps = {
+  isCollapsible: false,
+};
+
+const PillList = (props) => {
+  const {
+    ariaLabel,
+    children,
+    isCollapsible,
+  } = props;
+
+  const theme = React.useContext(ThemeContext);
+  const pillListRef = useRef();
+  const rollUpPillRef = useRef();
+  const [rollUpCount, setRollUpCount] = useState(0);
+  const [isRollUpRequired, setIsRollUpRequired] = useState(true);
+
+  const handleCollapse = () => {
+    let pillListHeight = pillListRef.current.getBoundingClientRect().height;
+    const pillHeight = rollUpPillRef.current.getBoundingClientRect().height + 12; // Margin + border width
+
+    if (pillListHeight <= pillHeight) {
+      setIsRollUpRequired(false);
+      return;
+    }
+
+    const childPills = pillListRef.current.querySelectorAll('[class*="pill-container"]');
+    let index = childPills.length - 1;
+    let pillCounter = 0;
+    while (pillListHeight > pillHeight) {
+      childPills[index].style.display = 'none';
+      index -= 1;
+      pillCounter += 1;
+      pillListHeight = pillListRef.current.getBoundingClientRect().height;
+    }
+    setIsRollUpRequired(true);
+    setRollUpCount(pillCounter);
+  };
+
+  useEffect(() => {
+    if (isCollapsible) {
+      handleCollapse();
+    }
+    pillListRef.current.style.visibility = 'visible';
+  }, []);
+
+  const handleRollUp = () => {
+    const childPills = pillListRef.current.querySelectorAll('[class*="pill-container"]');
+    let index = childPills.length - 1;
+    let pillCounter = rollUpCount;
+    while (pillCounter > 0) {
+      childPills[index].removeAttribute('style');
+      index -= 1;
+      pillCounter -= 1;
+    }
+    setIsRollUpRequired(false);
+    setRollUpCount(pillCounter);
+  };
+
+  const handleKeyDown = (event) => {
+    rollUpPillRef.current.setAttribute('data-terra-pills-show-focus-styles', 'true');
+    if (event.keyCode === KEY_RETURN || event.keyCode === KEY_SPACE) {
+      handleRollUp();
+    }
+  };
+
+  const handleMouseDown = () => {
+    rollUpPillRef.current.setAttribute('data-terra-pills-show-focus-styles', 'false');
+  };
+
+  let rollUpButton;
+  if (isCollapsible) {
```

Comment:
do we need to check this condition here for rendering the rollup button ?

Reply:
Yes, So how i have handled roll up is that based on the "isCollapsible" prop i am conditionally rendering the "Roll Up Pill" and the condition on line 130 is to check whether the "Pill List" is collapsed or expanded.